lib/action_handler.dart

```diff
@@ -107,20 +108,35 @@ class ActionHandler {
     params["message"] = message.text;
     params["tempGuid"] = message.guid;
 
-    SocketManager().sendMessage("send-message", params, (response) async {
-      String? tempGuid = message.guid;
+    VoidCallback sendSocketMessage = () {
+      SocketManager().sendMessage("send-message", params, (response) async {
+        String? tempGuid = message.guid;
 
-      // If there is an error, replace the temp value with an error
-      if (response['status'] != 200) {
-        message.guid = message.guid!.replaceAll("temp", "error-${response['error']['message']}");
-        message.error = response['status'] == 400 ? MessageError.BAD_REQUEST.code : MessageError.SERVER_ERROR.code;
+        // If there is an error, replace the temp value with an error
+        if (response['status'] != 200) {
+          message.guid = message.guid!.replaceAll("temp", "error-${response['error']['message']}");
+          message.error = response['status'] == 400 ? MessageError.BAD_REQUEST.code : MessageError.SERVER_ERROR.code;
 
-        await Message.replaceMessage(tempGuid, message);
-        NewMessageManager().updateMessage(chat, tempGuid!, message);
-      }
+          await Message.replaceMessage(tempGuid, message);
+          NewMessageManager().updateMessage(chat, tempGuid!, message);
+        }
 
-      completer.complete();
-    });
+        completer.complete();
+      });
+    };
+
+    bool isConnected = await InternetConnectionChecker().hasConnection;
+    if (!isConnected) {
+      StreamSubscription? sub;
```

Comment:
Should we add a timeout to this to cancel the subscription? I assume Apple does this as well. For instance, if it's been 1 minute and still no connection, cancel the message.

Reply:
Also, what if the user send the message (with no connection). Then leaves the app. If they come back 5 min later, will the message be timed-out? Or will it be "greyed out"? The correct answer should be that the message has an error

Reply:
Maybe we need a lifecycle checker for when the app is resumed, check for queued outgoing messages